ethereumj-core/src/main/java/org/ethereum/sharding/domain/Validator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.sharding.domain;
+
+import org.ethereum.util.ByteUtil;
+
+import static java.util.Arrays.copyOfRange;
+
+/**
+ * @author Mikhail Kalinin
+ * @since 21.07.2018
+ */
+public class Validator {
+
+    private static final int LOG_DATA_SIZE = 32 * 3 + 20; // pubKey + shardId + withdrawalAddress + randao
+
+    private byte[] pubKey;
+    private long withdrawalShard;
+    private byte[] withdrawalAddress;
+    private byte[] randao;
+
+    private Validator() {
+    }
+
+    public static Validator fromLogData(byte[] data) {
+        if (data.length < LOG_DATA_SIZE)
+            return null;
+
+        Validator v = new Validator();
+        v.pubKey = copyOfRange(data, 0, 32);
```

Comment:
Couldn't we use something like this https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/core/CallTransaction.java#L292 for initial parsing here? Current thing is like a magic

Reply:
Nice thing 👍 Will do